test/models/patient_test.rb

```diff
@@ -75,9 +78,34 @@ class PatientTest < ActiveSupport::TestCase
       assert @patient.valid?
       @patient.appointment_date = '2016-07-01'
       assert @patient.valid?
+      @patient.appointment_date = ( Date.today + 4 )
+      @patient.save
+      assert @patient.valid?
+      @patient2.appointment_date = ( Date.today + 8 )
+      assert @patient2.valid?
+      @patient2.save
     end
   end
 
+  describe 'pledge_summary' do
+      it "should return sent and remaining pledge for the next 7 days" do
+          summary = Patient.pledged_status_summary
+          assert_equal '{:pledged=>0, :sent=>0}', summary.to_s
+          [@patient, @patient2].each do |pt|
+            create :pregnancy, patient: pt, created_by: @user
+          end
+          @patient.pregnancy.dcaf_soft_pledge = 300
```

Comment:
slightly quicker one-liner: `@patient.pregnancy.update attribute: value`. That'll update the value and save it in one fell swoop, so you don't have to `.save` on a separate line!